pkg/scheduler/schedule_one.go

```diff
@@ -91,11 +91,35 @@ func (sched *Scheduler) scheduleOne(ctx context.Context) {
 
 	schedulingCycleCtx, cancel := context.WithCancel(ctx)
 	defer cancel()
-	scheduleResult, assumedPodInfo := sched.schedulingCycle(schedulingCycleCtx, state, fwk, podInfo, podsToActivate, start)
-	if scheduleResult.FeasibleNodes == 0 {
+
+	scheduleResult, assumedPodInfo, err := sched.schedulingCycle(schedulingCycleCtx, state, fwk, podInfo, start)
+	if err != nil {
```

Comment:
Moved the `metrics recording`, `unreserve plugins running`, `failureHandler` out of schedulingCycle.